scope/src/doctor/check.rs

```diff
@@ -544,7 +544,13 @@ impl GlobWalker for DefaultGlobWalker {
     ) -> Result<bool, RuntimeError> {
         for glob_str in paths {
             let glob_path = make_absolute(base_dir, glob_str);
-            for path in self.file_system.find_files(&glob_path)? {
+            let files = self.file_system.find_files(&glob_path)?;
+
+            if files.is_empty() {
+                return Ok(false);
```

Comment:
I don't see how this is different than the existing check to see if any of the files are newer.

```
if any cache out of date -> return false
```

In this case we're saying 

```
if any path is missing -> return false
```

You bring up an interesting point, but I believe the behavior is consistent.  
Perhaps slightly obfuscated by the use of loops instead of map/reduce, but consistent.  
_I think._

Reply:
Ah that's a good point, I had missed that.